src/datachain/data_storage/schema.py

```diff
@@ -213,7 +215,9 @@ def new_table(
     def get_table(self) -> "sa.Table":
         table = self.metadata.tables.get(self.name)
         if table is None:
-            sa.Table(self.name, self.metadata, autoload_with=self.engine)
+            if self.engine.is_closed:  # type: ignore[attr-defined]
+                self.engine._reconnect()  # type: ignore[attr-defined]
+            sa.Table(self.name, self.metadata, autoload_with=self.engine.engine)
```

Comment:
[C]: the only (?) place where we had SA engine directly used (usually we would use our wrapper that does reconnect on execute).

Internally `sa.Table` does some logic to query and fails if connection is closed.